server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -363,6 +363,66 @@ public void validate(IndexMode value, Map<Setting<?>, Object> settings) {
         Property.Final
     );
 
+    /**
+     * in time series mode, the start time of the index, timestamp must larger than start_time
+     * format is epoch_millis
+     */
+    public static final Setting<Long> TIME_SERIES_START_TIME = Setting.longSetting(
+        "index.time_series.start_time",
+        -1L,
+        -1L,
+        new Setting.Validator<>() {
+            @Override
+            public void validate(Long value) {}
+
+            @Override
+            public void validate(Long value, Map<Setting<?>, Object> settings) {
+                IndexMode mode = (IndexMode) settings.get(MODE);
+                if (mode != IndexMode.TIME_SERIES) {
+                    throw new IllegalArgumentException("index.time_series.start_time need to be used for time_series mode");
+                }
+            }
+
+            @Override
+            public Iterator<Setting<?>> settings() {
+                final List<Setting<?>> settings = List.of(MODE);
+                return settings.iterator();
+            }
+        },
+        Property.IndexScope,
+        Property.Dynamic
```

Comment:
I think we'd be ok if this one was `Property.Final` - I don't think we need to go back in time on this one. It's only the `end_time` that has to go forwards.